### PR TITLE
fix: markets table improvements

### PR DESCRIPTION
**Pull request by aforaleka**
markets table polish and fixes, especially on mobile: show volume (and sorted by that), truncate asset name in compact table + shorten pagination row so they don't overflow weridly
hide recently listed table when no markets are new (currently defined by having less than 7d sparklines historical data)

image
image

### CODE REVIEW

File: src/views/tables/MarketsTable.tsx [hunk 1]
```diff
@@ -263,7 +271,7 @@ export const MarketsTable = ({ className }: { className?: string }) => {
           navigate(`${AppRoute.Trade}/${market}`, { state: { from: AppRoute.Markets } })
         }
         defaultSortDescriptor={{
-          column: 'volume24H',
+          column: isTablet ? 'marketAndVolume' : 'volume24H',
```
Comment (aforaleka) [inline src/views/tables/MarketsTable.tsx:274]: unfortunately we don't re-sort on screen size change, only when it's first initialized. no fix for now

File: src/hooks/useMarketsData.ts [hunk 2]
```diff
@@ -261,28 +262,21 @@ export const useMarketsData = ({
...
-
   const marketFilters = useMemo(
     () =>
       [
-        MarketFilters.ALL,
-        showNewFilter ? MarketFilters.NEW : null,
-        testFlags.pml && MarketFilters.LAUNCHABLE,
         ...objectKeys(MARKET_FILTER_OPTIONS).filter((marketFilter) =>
-          markets.some((market) => market.tags?.some((tag) => tag === marketFilter))
+          markets.some((market) => filterFunctions[marketFilter](market))
```
Comment (jaredvu) [inline src/hooks/useMarketsData.ts:269]: 🙏

File: src/hooks/useMarketsData.ts [hunk 3]
```diff
@@ -261,28 +262,21 @@ export const useMarketsData = ({
...
         ),
-        hasPredictionMarkets && !testFlags.pml && MarketFilters.PREDICTION_MARKET_DEPRECATED,
       ].filter(isTruthy),
```
Comment (jaredvu) [inline src/hooks/useMarketsData.ts:271]: unnecessary filter now?

File: src/views/tables/MarketsTable.tsx [hunk 4]
```diff
@@ -77,7 +78,13 @@ export const MarketsTable = ({ className }: { className?: string }) => {
                   }}
                   name={name}
                   symbol={assetId}
-                />
+                >
+                  <Output
+                    type={OutputType.CompactFiat}
+                    value={volume24H ?? undefined}
```
Comment (jaredvu) [inline src/views/tables/MarketsTable.tsx:84]: does this need `??`? `<Output>` should handle both `null` and `undefined`?

**Review by jaredvu — APPROVED** [review]

**Pull request merged**